From 31bdc47f7af75e29cda91d222bca04b3b665bb64 Mon Sep 17 00:00:00 2001
From: GloriousEggroll <gloriouseggroll@gmail.com>
Date: Tue, 15 Apr 2025 14:47:48 -0600
Subject: [PATCH] revert manual dark/light mode changes, just use -symbolic
 icon from gtk to begin with, it autochanges dark/light

---
 data/dk.yumex.Yumex.gschema.xml.in            |   4 --
 data/gfx/dark_panel.png                       | Bin 24130 -> 0 bytes
 data/gfx/light_panel.png                      | Bin 21498 -> 0 bytes
 .../apps/yumex-update-dark-symbolic.svg       |  39 ------------------
 .../scalable/apps/yumex-update-symbolic.svg   |  39 ------------------
 data/icons/meson.build                        |   9 ----
 data/ui/preferences.blp                       |  13 ------
 docs/customize.md                             |  12 ------
 yumex/service/data.py                         |  15 ++-----
 yumex/ui/preferences.py                       |   4 --
 yumex/yumex_updater                           |   2 +-
 12 files changed, 4 insertions(+), 134 deletions(-)
 delete mode 100644 docs/customize.md

diff --git a/data/dk.yumex.Yumex.gschema.xml.in b/data/dk.yumex.Yumex.gschema.xml.in
index e0d8541..06ac4db 100644
--- a/data/dk.yumex.Yumex.gschema.xml.in
+++ b/data/dk.yumex.Yumex.gschema.xml.in
@@ -77,9 +77,5 @@
 			<default>true</default>
 			<summary>Send notification on updates</summary>
 		</key>
-		<key name="upd-dark-icon" type="b">
-			<default>false</default>
-			<summary>Use a dark updaer icon</summary>
-		</key>
 	</schema>
 </schemalist>
diff --git a/data/icons/meson.build b/data/icons/meson.build
index 17c9b2e..0841adf 100644
--- a/data/icons/meson.build
+++ b/data/icons/meson.build
@@ -3,12 +3,3 @@ install_data(
   join_paths(scalable_dir, ('@0@.svg').format(APPLICATION_ID)),
   install_dir: join_paths(get_option('datadir'), 'icons', scalable_dir),
 )
-install_data(
-  join_paths(scalable_dir, 'yumex-update-symbolic.svg'),
-  install_dir: join_paths(get_option('datadir'), 'icons', scalable_dir),
-)
-
-install_data(
-  join_paths(scalable_dir, 'yumex-update-dark-symbolic.svg'),
-  install_dir: join_paths(get_option('datadir'), 'icons', scalable_dir),
-)
\ No newline at end of file
diff --git a/data/ui/preferences.blp b/data/ui/preferences.blp
index 425cd14..021bb99 100644
--- a/data/ui/preferences.blp
+++ b/data/ui/preferences.blp
@@ -87,19 +87,6 @@ template $YumexPreferences: Adw.PreferencesDialog {
         }
       }
 
-      Adw.ActionRow {
-        title: _("Use dark systray icon");
-        visible: bind upd_show.active;
-
-        Gtk.Box {
-          orientation: horizontal;
-          vexpand: false;
-          valign: center;
-
-          Gtk.Switch upd_dark_icon {}
-        }
-      }
-
       Adw.ActionRow {
         title: _("Send Notification on updates");
 
diff --git a/docs/customize.md b/docs/customize.md
deleted file mode 100644
index 6318c39..0000000
--- a/docs/customize.md
+++ /dev/null
@@ -1,12 +0,0 @@
-## Switch the updater to use an dark icon
-
-Then using a display environment with light system tray, that don't support automatic changing colors on symbolic icons, like gnome
-you can set the yumex-updater to switch to a dark icon.
-
-```
-gsettings set dk.yumex.Yumex upd-dark-icon true
-systemctl --user restart yumex-updater
-```
-
-![light](../data/gfx/light_panel.png)
-![dark](../data/gfx/dark_panel.png)
diff --git a/yumex/service/data.py b/yumex/service/data.py
index f8f0cc9..322f927 100644
--- a/yumex/service/data.py
+++ b/yumex/service/data.py
@@ -48,7 +48,6 @@ class Config:
     show_icon: bool
     update_sync_interval: int
     send_notification: bool
-    dark_icon: bool
 
     @classmethod
     def from_gsettings(cls):
@@ -58,20 +57,17 @@ class Config:
         show_icon = settings.get_boolean("upd-show-icon")
         update_interval = settings.get_int("upd-interval")
         notification = settings.get_boolean("upd-notification")
-        dark_icon = settings.get_boolean("upd-dark-icon")
         logger.debug(f"CONFIG: custom_updater        = {custom_updater}")
         logger.debug(f"CONFIG: show_icon             = {show_icon}")
         logger.debug(f"CONFIG: update_sync_interval  = {update_interval}")
         logger.debug(f"CONFIG: send_notification     = {notification}")
-        logger.debug(f"CONFIG: dark_icon             = {dark_icon}")
-        return cls(custom_updater, show_icon, update_interval, notification, dark_icon)
+        return cls(custom_updater, show_icon, update_interval, notification)
 
 
 class Indicator:
-    def __init__(self, custom_updater, refresh_func, dark_icon):
+    def __init__(self, custom_updater, refresh_func):
         self._indicator = None
         self.custom_updater = custom_updater
-        self.dark_icon = dark_icon
         self.refresh_func = refresh_func
         self.last_pkgs = 0
         self.last_flatpaks = 0
@@ -88,14 +84,9 @@ class Indicator:
 
     def _factory(self):
         try:
-            if self.dark_icon:
-                logger.debug("Using dark icon")
-                icon_name = "yumex-update-dark-symbolic.svg"
-            else:
-                icon_name = "yumex-update-symbolic.svg"
             indicator = AppIndicator3.Indicator.new(
                 "System Update Monitor",
-                f"/usr/share/icons/hicolor/scalable/apps/{icon_name}",
+                "system-software-update-symbolic",
                 AppIndicator3.IndicatorCategory.APPLICATION_STATUS,
             )
             indicator.set_status(AppIndicator3.IndicatorStatus.ACTIVE)
diff --git a/yumex/ui/preferences.py b/yumex/ui/preferences.py
index c991143..4c5c1b7 100644
--- a/yumex/ui/preferences.py
+++ b/yumex/ui/preferences.py
@@ -38,7 +38,6 @@ class YumexPreferences(Adw.PreferencesDialog):
     upd_interval = Gtk.Template.Child()
     upd_show = Gtk.Template.Child()
     upd_notification = Gtk.Template.Child()
-    upd_dark_icon = Gtk.Template.Child()
 
     def __init__(self, presenter, **kwargs):
         super().__init__(**kwargs)
@@ -78,8 +77,6 @@ class YumexPreferences(Adw.PreferencesDialog):
         self.upd_show.set_state(self.settings.get_boolean("upd-show-icon"))
         self.upd_notification.set_active(self.settings.get_boolean("upd-notification"))
         self.upd_notification.set_state(self.settings.get_boolean("upd-notification"))
-        self.upd_dark_icon.set_active(self.settings.get_boolean("upd-dark-icon"))
-        self.upd_dark_icon.set_state(self.settings.get_boolean("upd-dark-icon"))
 
     def setup_metadata(self):
         period = self.settings.get_int("meta-load-periode")
@@ -129,7 +126,6 @@ class YumexPreferences(Adw.PreferencesDialog):
             pass
         self.settings.set_boolean("upd-show-icon", self.upd_show.get_state())
         self.settings.set_boolean("upd-notification", self.upd_notification.get_state())
-        self.settings.set_boolean("upd-dark-icon", self.upd_dark_icon.get_state())
         return location, remote
 
     def update_remote(self, current_location) -> str | None:
diff --git a/yumex/yumex_updater b/yumex/yumex_updater
index 3cdfa5e0..da7c9d5 100755
--- a/yumex/yumex_updater
+++ b/yumex/yumex_updater
@@ -99,7 +99,7 @@ def setup_notification():
 
 # Setup Global Constants
 CONFIG = Config.from_gsettings()
-INDICATOR = Indicator(custom_updater=CONFIG.custom_updater, refresh_func=refresh_updates, dark_icon=CONFIG.dark_icon)
+INDICATOR = Indicator(custom_updater=CONFIG.custom_updater, refresh_func=refresh_updates)
 
 if CONFIG.send_notification:
     NOTIFICATION: Notification = setup_notification()
-- 
2.49.0

